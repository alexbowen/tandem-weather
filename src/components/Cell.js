import React from 'react'
import { convertKelvinToCelsius, getTime } from './helpers/displayUtils'

const Cell = ({ time, temperature, weather }) => (
    <div className="forecast-cell">
        <span><img src={`http://openweathermap.org/img/w/${weather.icon}.png`} height="50px" width="50px" alt={weather.description} /></span>
        <span>{getTime(time)}</span>
        <span>{convertKelvinToCelsius(temperature)}&deg;C</span>
    </div>
)

export default Cell
