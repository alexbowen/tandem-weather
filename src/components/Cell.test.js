import React from 'react'
import ReactDOM from 'react-dom'
import Cell from './Cell'
import renderer from 'react-test-renderer'

test('Cell component', () => {

    const component = renderer.create(
        <Cell messages={{time:"2018-02-22 00:00:00", temperature: 274.72}} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})