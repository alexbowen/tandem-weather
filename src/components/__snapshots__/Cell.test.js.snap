// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component 1`] = `
<div
  className="forecast-cell"
>
  <span>
    Invalid Date
  </span>
  <span>
    NaN
    Â°C
  </span>
</div>
`;
